### DISCUSSION TITLE
新的配置文件极验的内容填什么

### DISCUSSION THREAD

**Discussion by PsalmReimu** [0-8-4/miui-auto-tasks#162] [问与答｜Q&A]
preference:
    geetest_url: ''
    geetest_params: {}
    geetest_data: {}

这个怎么填啊，有没有参考

**Comment by PsalmReimu** [comment]
![屏幕截图(48)](https://github.com/0-8-4/miui-auto-tasks/assets/22426441/437714d0-368b-4014-92da-313910b1f060)

**Reply by BA7JCM** [reply]
看到他[config.py里面第80到86行](https://github.com/0-8-4/miui-auto-tasks/blob/master/utils/config.py#L80-L86)写的
```python
class Preference(BaseModel):
    geetest_url: str = ""
    """极验验证URL"""
    geetest_params: Dict = {}
    """极验自定义params参数"""
    geetest_data: Dict = {}
    """极验自定义data参数"""
```
然后我查了下极验证好像自己是搞人机验证的，没弄明白要干啥

**Comment by Night-stars-1** [comment]
不知道这个是干嘛的说明你暂时解决不了这个问题
建议不要纠结这个

变量说明是故意写的模糊不清的

**Comment by TaiMiao** [comment]


**Reply by a312354805** [reply]
接入2Captcha就行了 我都跑了一年多了

**Reply by SeventeenOvO** [reply]
您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。

**Reply by slayercjc** [reply]
请大神教一下

**Reply by a312354805** [reply]
1.pip3 install 2captcha-python
2.
![image](https://github.com/user-attachments/assets/55d510f9-5ca1-4b1a-aa6a-957491958333)
3.
![image](https://github.com/user-attachments/assets/70eb755b-0a76-48ce-b157-634aadd13778)

还需要 

import json


需要注意的是 userAgent 的值  需要跟utils.py上面的headers里面的一样，如果你们用项目里面那个头不行（因为我用那个老是验证结果失败）， 可以用我这个 Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0

_conf.preference.api_key 这个地方你可以自己写死 或者 你自己读配置里面的

**Reply by a312354805** [reply]
> 您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。

试试看 我发的这个

**Reply by slayercjc** [reply]
> > 您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。
> 
> 试试看 我发的这个

pip3 install 2captcha-python   大佬我是小白，请问这个命令在哪运行呢？

**Reply by a312354805** [reply]
> > > 您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。
> > 
> > 
> > 试试看 我发的这个
> 
> pip3 install 2captcha-python 大佬我是小白，请问这个命令在哪运行呢？

你装了python 在命令行里面运行就行了撒 ， 怎么跑的

**Reply by slayercjc** [reply]
> > > > 您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。
> > > 
> > > 
> > > 试试看 我发的这个
> > 
> > 
> > pip3 install 2captcha-python 大佬我是小白，请问这个命令在哪运行呢？
> 
> 你装了python 在命令行里面运行就行了撒 ， 怎么跑的

我是用青龙跑的

**Reply by slayercjc** [reply]
> > > > > 您好，2Captcha这个对接的参数总是填不对，能否参考一下您的这三个参数。
> > > > 
> > > > 
> > > > 试试看 我发的这个
> > > 
> > > 
> > > pip3 install 2captcha-python 大佬我是小白，请问这个命令在哪运行呢？
> > 
> > 
> > 你装了python 在命令行里面运行就行了撒 ， 怎么跑的
> 
> 我是用青龙跑的

大佬有TG，想TG交流下。。。

**Reply by a312354805** [reply]
依赖管理里面创建一个python3的依赖 名称 2captcha-python

**Reply by slayercjc** [reply]
> 依赖管理里面创建一个python3的依赖 名称 2captcha-python
请教下 url填写啥呢

**Reply by a312354805** [reply]
看我截图！！！！按照我那个来就行了，基本上可以不在配置里面配置，你自己写死就行了   把  _conf.preference.api_key换成你的秘钥  放在双引号里面就行了  "秘钥"

**Reply by slayercjc** [reply]
> 看我截图！！！！按照我那个来就行了，基本上可以不在配置里面配置，你自己写死就行了 把 _conf.preference.api_key换成你的秘钥 放在双引号里面就行了 "秘钥"
![4449-21](https://github.com/user-attachments/assets/e5c72532-24ca-434b-8b68-7fb75f99b7b9)

我的这个配置貌似和你的不一样。。。

**Reply by a312354805** [reply]
"""获取人机验证结果"""
    try:
        validate = None
        log.error("开始使用[2Captcha]解密验证码")
        solver = TwoCaptcha("秘钥")
        geetest_data = solver.geetest(gt=gt,
                              apiServer='api.geetest.com',
                              challenge=challenge,userAgent='Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0',
                              url=url)
            # 解析 code 字段为字典对象
        code_data = json.loads(geetest_data['code'])
            # 获取 geetest_challenge 和 geetest_validate 的值
            #challenge = code_data['geetest_challenge']
        validate = code_data['geetest_validate']
        log.error("获取解密参数成功")
        return GeetestResult(challenge=challenge, validate=validate)
    except Exception:  # pylint: disable=broad-exception-caught
        log.exception("获取人机验证结果异常")
        return GeetestResult(challenge="", validate="")

替换掉你的 然后看我截图两个文件 圈出来的地方 你自己加上

**Reply by slayercjc** [reply]
> """获取人机验证结果""" try: validate = None log.error("开始使用[2Captcha]解密验证码") solver = TwoCaptcha("秘钥") geetest_data = solver.geetest(gt=gt, apiServer='api.geetest.com', challenge=challenge,userAgent='Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0', url=url) # 解析 code 字段为字典对象 code_data = json.loads(geetest_data['code']) # 获取 geetest_challenge 和 geetest_validate 的值 #challenge = code_data['geetest_challenge'] validate = code_data['geetest_validate'] log.error("获取解密参数成功") return GeetestResult(challenge=challenge, validate=validate) except Exception: # pylint: disable=broad-exception-caught log.exception("获取人机验证结果异常") return GeetestResult(challenge="", validate="")
> 
> 替换掉你的 然后看我截图两个文件 圈出来的地方 你自己加上
好的，谢谢，solver = TwoCaptcha("秘钥") 秘钥就是注册2Captcha的API密钥是吧？

**Reply by slayercjc** [reply]
> """获取人机验证结果""" try: validate = None log.error("开始使用[2Captcha]解密验证码") solver = TwoCaptcha("秘钥") geetest_data = solver.geetest(gt=gt, apiServer='api.geetest.com', challenge=challenge,userAgent='Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0', url=url) # 解析 code 字段为字典对象 code_data = json.loads(geetest_data['code']) # 获取 geetest_challenge 和 geetest_validate 的值 #challenge = code_data['geetest_challenge'] validate = code_data['geetest_validate'] log.error("获取解密参数成功") return GeetestResult(challenge=challenge, validate=validate) except Exception: # pylint: disable=broad-exception-caught log.exception("获取人机验证结果异常") return GeetestResult(challenge="", validate="")
> 
> 替换掉你的 然后看我截图两个文件 圈出来的地方 你自己加上

Traceback (most recent call last):
  File "/ql/data/scripts/0-8-4_miui-auto-tasks_master/miuitask.py", line 11, in <module>
    from utils.api.login import Login
  File "/ql/data/scripts/0-8-4_miui-auto-tasks_master/utils/api/login.py", line 17, in <module>
    from ..utils import generate_qrcode
  File "/ql/data/scripts/0-8-4_miui-auto-tasks_master/utils/utils.py", line 18, in <module>
    from .captcha import get_validate
  File "/ql/data/scripts/0-8-4_miui-auto-tasks_master/utils/captcha.py", line 35
    code_data = json.loads(geetest_data['code'])
    ^^^^^^^^^
SyntaxError: expected 'except' or 'finally' block

大佬我改完运行 提示这个错误了。。。

**Reply by lwh041009** [reply]
请问现在还能获取token吗，接入了打码狗，还是无法获取token，是不是加密变了

**Reply by a312354805** [reply]
> 请问现在还能获取token吗，接入了打码狗，还是无法获取token，是不是加密变了

我刚刚看了一下

20:12:32 - INFO  | system_info.print_info:13 - MIUI-AUTO-TASK v1.7.5
20:12:32 - INFO  | system_info.print_info:14 - ---------- 系统信息 -------------
20:12:32 - INFO  | system_info.system_info:18 - 操作系统平台: Linux-6.6.34-9.oc9.x86_64-x86_64-with
20:12:32 - INFO  | system_info.system_info:19 - 操作系统版本: #1 SMP PREEMPT_DYNAMIC Wed Jun 19 19:35:45 CST 2024
20:12:32 - INFO  | system_info.system_info:20 - 操作系统名称: Linux
20:12:32 - INFO  | system_info.system_info:21 - 操作系统位元: ('64bit', '')
20:12:32 - INFO  | system_info.system_info:22 - 操作系统类型: x86_64
20:12:32 - INFO  | system_info.system_info:23 - 处理器信息: 
20:12:32 - INFO  | system_info.system_info:24 - Python 版本: 3.11.10 ('main', 'Oct 19 2024 00:56:54')
20:12:33 - INFO  | login.login:92 - Cookie无效，重新复写
20:12:33 - INFO  | login.checkin_info:285 - 双重惊喜：每天可在小米社区微信小程序额外签到一次
20:12:33 - INFO  | miuitask.main:36 - 开始执行升级小米社区App至最新版本任务
20:12:33 - ERROR | miuitask.main:43 - 未找到升级小米社区App至最新版本任务
20:12:33 - INFO  | miuitask.main:36 - 开始执行每日签到任务
20:12:33 - INFO  | miuitask.main:38 - 每日签到任务已完成
20:12:33 - INFO  | miuitask.main:36 - 开始执行浏览帖子超过10秒任务
20:12:34 - SUCCESS | sign.sign:144 - 浏览帖子超过10秒结果: 成长值+1
20:12:34 - INFO  | miuitask.main:36 - 开始执行小米社区微信小程序签到获成长值任务
20:12:34 - ERROR | miuitask.main:43 - 未找到小米社区微信小程序签到获成长值任务
20:12:34 - INFO  | miuitask.main:36 - 开始执行参与拔萝卜获得奖励任务
20:12:34 - SUCCESS | sign.sign:146 - 参与拔萝卜获得奖励结果: ok
20:12:34 - INFO  | miuitask.main:57 - 资深成员 成长值: 3055

**Reply by lwh041009** [reply]
还是作者这个源码吗，怎么我获取token失败了

**Reply by lwh041009** [reply]
请问你的二维码扫码有用吗，我的登录二维码扫了之后一点反应也没有

**Reply by a312354805** [reply]
> 请问那个带_a的cookie咋抓啊，我问ai他说玩到app抓

<img width="955" height="668" alt="image" src="https://github.com/user-attachments/assets/858c5c82-c1d8-46aa-b37f-eaf3105b898c" />
跑了一年多，一直没问题，不是很清楚你那个什么情况，没有用过扫码登陆，也没说要抓扫码东西啊，如果要验证手机号 用浏览器登陆一下，用它的userAgent就行了吧

**Reply by lwh041009** [reply]
请问你是跑在哪里啊，我是跑在Linux服务器上

**Reply by a312354805** [reply]
> 请问你是跑在哪里啊，我是跑在Linux服务器上

建议你去研究一下这个issues，https://github.com/0-8-4/miui-auto-tasks/issues/139#issuecomment-1822502989

**Reply by lwh041009** [reply]
目前能获取token，但是cookie过期怎么解决，不想老是自己抓包获取

**Reply by a312354805** [reply]
> 目前能获取token，但是cookie过期怎么解决，不想老是自己抓包获取

啊？程序不是自动重新登陆吗？从来没有去抓过包啊

**Reply by lwh041009** [reply]
这样嘛，那我再看看，谢谢啦

**Reply by lwh041009** [reply]
能问问你为啥能完成微信小程序的任务吗

**Reply by a312354805** [reply]
> 能问问你为啥能完成微信小程序的任务吗

我的版本是老版本的，没有微信签到的任务，前几天来用AI把新的集成进去了，然后运行任务就自动签到了，都不需要做啥。如果你需要可以把我这套跑的代码发一份给你，留个邮箱

**Reply by a312354805** [reply]
> > 能问问你为啥能完成微信小程序的任务吗
> 
> 我的版本是老版本的，没有微信签到的任务，前几天来用AI把新的集成进去了，然后运行任务就自动签到了，都不需要做啥。如果你需要可以把我这套跑的代码发一份给你，留个邮箱

https://github.com/a312354805/By-MIUI-Auto-Tasks-2026.git 如果有需要可以看一下这个，改动不多

**Reply by lwh041009** [reply]
好的，谢谢
